ambry-commons/src/main/java/com.github.ambry.commons/ByteBufferReadableStreamChannel.java

```diff
@@ -30,16 +33,28 @@
 public class ByteBufferReadableStreamChannel implements ReadableStreamChannel {
   private final AtomicBoolean channelOpen = new AtomicBoolean(true);
   private final AtomicBoolean channelEmptied = new AtomicBoolean(false);
-  private final ByteBuffer buffer;
+  private final List<ByteBuffer> buffers;
   private final int size;
 
   /**
    * Constructs a {@link ReadableStreamChannel} whose read operations return data from the provided {@code buffer}.
    * @param buffer the {@link ByteBuffer} that is used to retrieve data from on invocation of read operations.
    */
   public ByteBufferReadableStreamChannel(ByteBuffer buffer) {
-    this.buffer = buffer;
-    size = buffer.remaining();
+    this(Arrays.asList(buffer));
+  }
+
+  /**
+   * Constructs a {@link ReadableStreamChannel} whose read operations return data from the provided {@code bufferList}.
+   * @param bufferList list of {@link ByteBuffer} that is used to retrieve data from on invocation of read operations.
+   */
+  public ByteBufferReadableStreamChannel(List<ByteBuffer> bufferList) {
```

Comment:
@vgkholla @moontails what did you decide about also letting this class take in buffers after construction?